lib/datadog/tracing/distributed/baggage.rb

```diff
@@ -125,6 +130,33 @@ def parse_baggage_header(baggage_header)
           end
           baggage
         end
+
+        # Convert baggage items to trace_distributed_tags with 'baggage.' prefix
+        # based on the baggage_tag_keys configuration
+        def build_baggage_tags(baggage)
+          return {} if baggage.empty?
+
+          # Get the configuration for which baggage keys should become span tags
+          baggage_tag_keys = ::Datadog.configuration.tracing.baggage_tag_keys
```

Comment:
Let's move this to the `initializer` method instead, and cache the result:
```suggestion
          baggage_tag_keys = @baggage_tag_keys
```